python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -638,6 +639,11 @@ def from_rule_evaluation_results(
         num_discarded: int,
         dynamic_partitions_store: "DynamicPartitionsStore",
     ) -> "AutoMaterializeAssetEvaluation":
+        rule_snapshots = []
+        auto_materialize_policies = asset_graph.auto_materialize_policies_by_key.get(asset_key)
+        if auto_materialize_policies:
```

Comment:
will there never not be a policy? seems like we'd want to raise a check error in that case?

Reply:
A lot of the old reconciliation scenario tests don't contain explicit auto materialize policies on the assets, and a check would cause a failure for all of those tests. 

I think for now we can rely on this function only being called for assets containing an auto-materialize policy, until we can migrate all of the tests to have auto materialize policies

Reply:
Not the end of the world, but I don't love having code that's more complicated to support the tests.

How difficult would it be to update the tests? E.g. just have something in the test harness that adds an eager policy to all of the assets (and maybe a lazy policy instead for all of the assets in freshness_policy_scenarios.py).

Reply:
It's surprisingly tricky, I've spent a significant amount of time today to get these tests to pass but I think i'm almost there. Turns out there's a bunch of edge cases (i.e. some assets in a certain reconciliation scenario have AMPs, perf tests rely on snapshots, etc.)

Still need to look into additional test failures, i.e. this case is emitting an auto-materialize asset evaluation for asset1 even though it doesn't have a policy. Is it expected for an evaluation to exist here?: https://github.com/dagster-io/dagster/blob/652583c7995e059bb0936f01ab1abc1ce5c4cb26/python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/scenarios/auto_materialize_policy_scenarios.py#L58-L69

Reply:
Ah, looks like we auto-target all assets regardless of whether they have auto materialize policies, so it is expected that each asset has an evaluation